shadowsocks/udp_buffer_pool.go

```diff
@@ -0,0 +1,22 @@
+package shadowsocks
+
+import "sync"
+
+// maxUDPBufferSize is the maximum UDP packet size in bytes.
+const maxUDPBufferSize = 16 * 1024
+
+var pool = sync.Pool{
+	New: func() interface{} {
+		return make([]byte, maxUDPBufferSize)
```

Comment:
It does worry me that `bytes.Buffer`s can grow unboundedly. Since we're only using these buffers for UDP (read: fixed max size), I believe we're better off with slices.